src/SFML/Window/WindowBase.cpp

```diff
@@ -95,7 +95,17 @@ void WindowBase::create(VideoMode mode, const String& title, std::uint32_t style
     WindowBase::create(mode, style, state);
 
     // Recreate the window implementation
-    m_impl = priv::WindowImpl::create(mode, title, style, state, ContextSettings(0, 0, 0, 0, 0, 0xFFFFFFFF, false));
+    m_impl = priv::WindowImpl::create(mode,
+                                      title,
+                                      style,
+                                      state,
+                                      ContextSettings{/* depthBits */ 0,
+                                                      /* stencilBits */ 0,
+                                                      /* antialiasingLevel */ 0,
+                                                      /* majorVersion */ 0,
+                                                      /* minorVersion */ 0,
+                                                      /* attributeFlags */ 0xFFFFFFFF,
+                                                      /* sRgbCapable */ false});
```

Comment:
Not sure, I'm a fan of this.

It doesn't look particularly nice and is rather brittle. Renaming will not update the comments, changing the field order will lead to silent failures. We're making the filed order part of the public API.

Reply:
From the code above, I have no idea what all the specified values mean unless I go and check the definition of `ContextSettings` and I carefully match the arguments provided here with the fields of `ContextSettings` by manually counting.

Adding the comments provides clarity on what these values actually mean. In C++20, designated initializers would be even better, but we're targeting C++17.

> Renaming will not update the comments, changing the field order will lead to silent failures. We're making the filed order part of the public API.

These arguments are all extremely weak. The field order and the names of the field **are** part of the public API, because `ContextSettings` is now an aggregate!

If we ever decide to change the name of a field or the field order, keeping some comments up-to-date is going to be the least of our worries because we will be breaking existing users and potentially changing run-time behavior silently, without causing a compilation failure.

Reply:
> Adding the comments provides clarity on what these values actually mean. In C++20, designated initializers would be even better, but we're targeting C++17.

And why is this needed now and not before with a constructor?

> These arguments are not compelling. The field order and the names of the field are part of the public API, because `ContextSettings` is now an aggregate!

By pointing out that I'm not sure the field order should be part of the API, I'm of course indirectly challenging whether it should be an aggregate...

Are there any benefits beyond just having a different way to construct it?

Reply:
> And why is this needed now and not before with a constructor?

It's not needed now, I would have also done the same if we kept the constructor. It's mostly a drive-by small change that I realized would improve readability as I was adapting the existing uses of `ContextSettings` to being an aggregate.

> By pointing out that I'm not sure the field order should be part of the API, I'm of course indirectly challenging whether it should be an aggregate...

Sorry, it wasn't clear because you specifically selected the `/* ... */` comments in your review and the rest of your feedback was directed specifically to the comments.

> Are there any benefits beyond just having a different way to construct it?

Yes: C++20 projects can use designated initializers, aggregates can be automatically destructured and reflected on, the implementation is much simpler.

Prior to this PR `ContextSettings` was essentially used like an aggregate without being one, as its constructor had all its arguments defaulted. 

Please let me know if you're OK with moving forward before merging.

Reply:
I take a bit issue with doing a change with the main argument that someone using the latest compile could potentially benefit from it.

Personally, I don't see something wrong with having a constructor on an aggregate type.

But I'm okay with the change.

Reply:
> I take a bit issue with doing a change with the main argument that someone using the latest compile could potentially benefit from it.

Perhaps I was not very clear -- what I meant to say is that making `ContextSettings` an aggregate has the extra benefit of allowing users targeting C++20 to use designated initializers, but most importantly it's a simplification of our implementation and API. 
 
> Personally, I don't see something wrong with having a constructor on an aggregate type.

An aggregate type, by definition, cannot have any user-declared constructor.